### DISCUSSION TITLE
Disable my own Namespace for Auto-Completion (Apex)

### DISCUSSION THREAD

**Discussion by Szandor72** [forcedotcom/salesforcedx-vscode#5471] [Ideas]
**Describe the solution you'd like**
As an ISV we work with namespaces all of the time. In VS Code, if I want to use the autocomplete feature, autocompletion looks like this

![completion with namespace](https://i.imgur.com/5uWukOL.png)

Is there a way to tell VS Code or the Salesforce Extension not to use namespace? 

The omitted Namespace should be the one defined in the scratch project definition or scratch config file so to avoid confusion when working on a package extension.

**Comment by vazexqi** [comment]
Just to be clear, those are not _just_ namespaces. Those are custom objects from other namespaces. So, they could be used, and referenced. 

There are at least two ways we can handle this:

1. We could move this to the bottom of the list (if there are other completion suggestions)
2. We could do provide an option like what you suggest (hide namespace completions suggestions). However, how would you _discover_ those items then if you need to use them? I'm not sure if this is possible, but _maybe_ hitting ctrl+space twice would be the action to explicitly suggest things from other namespaces)?

Thoughts?

This is related (but not identical) to https://github.com/forcedotcom/salesforcedx-vscode/issues/399 since these are real entities that you can reference. You get them after you invoke "SFDX: Refresh SObject Definitions".

**Comment by Szandor72** [comment]
I agree with your analysis. 

Hence I suggested to strip only(!) the namespace that is configured in one of the config json files. No way should we strip Namespaces generally.

Maybe I misunderstood something?

I changed the title of the issue to better reflect my suggestion.

**Comment by AnanyaJha** [comment]
Hi @Szandor72! It's been a few years, but curious to see if this is still a reoccurring issue for you?

**Comment by AnanyaJha** [comment]
Hi @Szandor72 - following up again a few years later :) Wanted to see if this still an issue or feature request you're interested in. Otherwise, we'll close this as not addressed for now. Thanks so much!

**Comment by Szandor72** [comment]
Generally, yes - as an opt in toggle somewhere in settings. 

My weighting has changed, though. It moved from "Need" to "Nice to have". I do apologize for not having followed up the first time.
